Sources/xcresultparser/CoberturaCoverageConverter.swift

```diff
@@ -60,44 +122,55 @@ public class CoberturaCoverageConverter: CoverageConverter, XmlSerializable {
         rootElement.addChild(packagesElement)
 
+        // Get the xccov results as a JSON.
+        var arguments = ["xccov", "view"]
+        if resultFile.url.pathExtension == "xcresult" {
+            arguments.append("--archive")
+        }
+        arguments.append("--json")
+        arguments.append(resultFile.url.path)
+        let coverageData = try Shell.execute(program: "/usr/bin/xcrun", with: arguments)
+        let resultsString = String(decoding: coverageData, as: UTF8.self)
+        guard let jsonData = resultsString.data(using: String.Encoding.utf8) else {
+            writeToStdError("Failed to convert to Data")
+            throw JSONParseError.convertError(code: 0, message: "Failed to convert to Data object")
+        }
+        guard let coverageJson = try? JSONDecoder().decode(FileCoverage.self, from: jsonData) else {
+            writeToStdError("Failed to convert to JSON")
+            throw JSONParseError.convertError(code: 0, message: "Failed to convert to JSON Object")
+        }
+
         let fileInfoSemaphore = DispatchSemaphore(value: 1)
         var fileInfo: [FileInfo] = []
-        
-        // since we need to invoke xccov for each file, it takes pretty much time
-        // so we invoke it in parallel on 8 threads, that speeds up things considerably
         let queue = OperationQueue()
-        queue.maxConcurrentOperationCount = 8 //Deadlock if this is = 1
+        queue.maxConcurrentOperationCount = ProcessInfo.processInfo.processorCount //Deadlock if this is = 1
         queue.qualityOfService = .userInitiated
+        for (fileName, value) in coverageJson.files {
+            let op = BlockOperation {
+                var fileLines = [LineInfo]() // This will store information about each line.
 
-        var processedFiles = [String]()
-        for target in codeCoverage.targets {
-            if !coverageTargets.isEmpty {
-                guard coverageTargets.contains(target.name) else { continue }
-            }
-            for codeCovFile in target.files {
-                let file = codeCovFile.path
-                guard !processedFiles.contains(file) else { continue }
-                processedFiles.append(file)
-                guard !file.isEmpty else { continue }
-                if !quiet {
-                    writeToStdError("Coverage for: \(file)\n")
-                }
-                let op = BlockOperation { [self] in
-                    do {
-                        let coverage = try fileCoverage(for: file, relativeTo: projectRoot)
-                        fileInfoSemaphore.wait()
-                        fileInfo.append(coverage)
-                        fileInfoSemaphore.signal()
-                    } catch {
-                        writeToStdErrorLn(error.localizedDescription)
+                for lineData in value {
+                    let lineNum = lineData.line
+                    var covered = lineData.executionCount
+                    if covered != nil {
```

Comment:
Since we're not doing anything if `covered` == `nil`, could we switch this to a `guard` statement? It'd eliminate the need for force unwrapping `covered` below as well.
```suggestion
                    let lineNum = lineData.line
                    guard var covered == lineData.executionCount else {
                    ...
```